Name.lastIndexOf is missing in Java 22

I tried to compile against Java 22 and it failed with 

>  [INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.7-SNAPSHOT:compile (default-compile) on project org.eclipse.jdt.core.javac: Compilation failure: Compilation failure:
[ERROR] /Users/fbricon/Dev/projects/eclipse-jdt-core-incubator/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java:[1252,36] cannot find symbol
[ERROR]   symbol:   method lastIndexOf(byte)
[ERROR]   location: variable javac of type com.sun.tools.javac.util.Name
[ERROR] -> [Help 1]

See 
https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/blob/ef2bc4e5305e795ac60fa609f14061fabe7fce8c/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java#L1252

I seem to have fixed it with a simple change:

```java
private Name convert(com.sun.tools.javac.util.Name javac) {
		if (javac == null || Objects.equals(javac, Names.instance(this.context).error) || Objects.equals(javac, Names.instance(this.context).empty)) {
			return null;
		}
		String nameString = javac.toString();
		int lastDot = nameString.lastIndexOf(".");
		if (lastDot < 0) {
			return this.ast.newSimpleName(nameString);
		} else {
			return this.ast.newQualifiedName(convert(javac.subName(0, lastDot)), (SimpleName)convert(javac.subName(lastDot + 1, javac.length() - 1)));
		}
		// position is set later, in FixPositions, as computing them depends on the sibling
	}
```

--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java
@@ -1411,9 +1411,10 @@ class JavacConverter {
 		if (javac == null || Objects.equals(javac, Names.instance(this.context).error) || Objects.equals(javac, Names.instance(this.context).empty)) {
 			return null;
 		}
-		int lastDot = javac.lastIndexOf((byte)'.');
+		String nameString = javac.toString();
+		int lastDot = nameString.lastIndexOf(".");
 		if (lastDot < 0) {
-			return this.ast.newSimpleName(javac.toString());
+			return this.ast.newSimpleName(nameString);
 		} else {
 			return this.ast.newQualifiedName(convert(javac.subName(0, lastDot)), (SimpleName)convert(javac.subName(lastDot + 1, javac.length() - 1)));
 		}

